Userland/Libraries/LibWeb/Layout/TableFormattingContext.cpp

```diff
@@ -241,10 +241,12 @@ void TableFormattingContext::compute_cell_measures()
         CSSPixels border_left = use_collapsing_borders_model ? round(cell_state.border_left / 2) : computed_values.border_left().width;
         CSSPixels border_right = use_collapsing_borders_model ? round(cell_state.border_right / 2) : computed_values.border_right().width;
 
-        auto min_content_width = calculate_min_content_width(cell.box);
-        auto max_content_width = calculate_max_content_width(cell.box);
-        auto min_content_height = calculate_min_content_height(cell.box, max_content_width);
-        auto max_content_height = calculate_max_content_height(cell.box, min_content_width);
+        auto sizes_for_min_content_width = calculate_sizes_for_min_content_width_box(cell.box);
+        auto min_content_width = sizes_for_min_content_width.width;
+        auto sizes_for_max_content_width = calculate_sizes_for_max_content_width_box(cell.box);
+        auto max_content_width = sizes_for_max_content_width.width;
+        auto min_content_height = sizes_for_max_content_width.height.has_value() ? sizes_for_max_content_width.height.value() : calculate_min_content_height(cell.box, max_content_width);
```

Comment:
that looks very confusing. in what cases `calculate_sizes_for_max_content_width_box()` does return height? why can't we make `calculate_min_content_height()` always work instead?

Reply:
SVGs missing natural width / height are the only case for which this ternary is required. It is a bit annoying, for sure.